brontide/noise.go

```diff
@@ -68,8 +68,22 @@ var (
 		pool.DefaultReadBufferGCInterval,
 		pool.DefaultReadBufferExpiryInterval,
 	)
+
+	// readPool is a package level ReadPool that will be used by default to
+	// read messages from brontide connections.
+	readPool = pool.NewRead(
+		readBufferPool, runtime.NumCPU(), pool.DefaultWorkerTimeout,
+	)
 )
 
+func init() {
+	// TODO(conner): pass read pool to listeners and dialers to avoid
+	// pkg-level read pool
+	if err := readPool.Start(); err != nil {
```

Comment:
So then the pool can never be stopped once the package is imported?

Reply:
yeah, not my favorite thing. my longer term solution is to initialize listeners and create a new `Dialer` object that has the same semantics of `brontide.Dial`, but which can get its own read pool. then the pool can be started stopped w/ more control. This would be a bigger change, but happy to make that